### PR TITLE
Implement website features to main branch

### PR DESCRIPTION
**Pull request by rbhatt1999**
<h1>Video presentation</h1>

 [View with Zoom](https://drive.google.com/file/d/1CHIy6JadeflqTHL7fnzZqRLDZ-NjmveH/view?usp=share_link)

<h1>Live version</h1>

[Website Live Preview](https://rbhatt1999.github.io/JavaScript-Capstone/dist/)

### In this Pull request, we did the following :
<hr>

<h1>Build these interfaces:</h1>

- The home page.
- The comments popup.

<h2>Home page</h2>
When the page loads, the web app retrieves data from:

- The selected API shows the list of items on the screen.
- The Involvement API to show the item likes.
- the page makes 2 requests:
- One to the base API.
- And one to the Involvement API.

When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is updated.

- When the user clicks on the "Comments" button, the Comments popup appears.
- Home page header and navigation similar to the given mockup.
- Home page footer similar to the given mockup.

<h2>Comments popup</h2>

When the popup loads, the web app retrieves data from:

- The selected API shows details about the selected item.
- The Involvement API to show the item comments.
- When the user clicks on the "Comment" button, the data is recorded in the Involvement API and the screen is updated.

### CODE REVIEW

File: src/test/functions.test.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    countProducts(array);
+    expect(countProducts(array)).toEqual(2);
+  });
+
+  test('Count Comments', () => {
+    const array = ['item1', 'item2'];
+    calculateComments(array);
+    expect(calculateComments(array)).toEqual('Total Comments(2)');
+  });
+});
```
Comment (muneebulrehman) [inline src/test/functions.test.js]: One of the project requirements was to test the functions which calculate the count of things. You did a great job testing the `countProducts` and `calculateComments` functions. However, you forgot to test the **count for the likes**. As you have the count for the likes in the UI, kindly include the test for this functionality also.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Good code structure
- Good design

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/test/functions.test.js (hunk 1 above)
Comment (rbhatt1999) [reply src/test/functions.test.js]: create a test for count likes

**Review by fobadara — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Linters are passing
✅ Project looks professional

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**